src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -8986,6 +8986,19 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
                         if( func_subrout->type == ASR::symbolType::Function ) {
                             ASR::Function_t* func = down_cast<ASR::Function_t>(func_subrout);
                             orig_arg = ASRUtils::EXPR2VAR(func->m_args[i]);
+                        } else if (func_subrout->type == ASR::symbolType::ClassProcedure) {
+                            ASR::ClassProcedure_t* class_proc = down_cast<ASR::ClassProcedure_t>(func_subrout);
+                            ASR::symbol_t* func_sym = class_proc->m_proc;
+                            if (func_sym->type == ASR::symbolType::Function) {
```

Comment:
```suggestion
                            if (ASR::is_a<ASR::Function_t>(*func_sym)) {
```
And the same everywhere else around here.

If you could please fix that in the next PR, that would be great.